### PR TITLE
feat: support client certificate verification

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by membphis — APPROVED** [review]

File: apisix/ssl/router/radixtree_sni.lua [hunk 1]
```diff
@@ -194,6 +195,24 @@ function _M.match_and_set(api_ctx)
         end
     end
 
+    if matched_ssl.value.client then
+        local client_cert = matched_ssl.value.client.ca
```
Comment (tokers) [inline apisix/ssl/router/radixtree_sni.lua:199]: The name `client_cert` is confusing, actually, it's the CA cert(s) to verify the client cert. What about `ca_cert`.

File: apisix/schema_def.lua [hunk 2]
```diff
@@ -642,6 +642,18 @@ _M.ssl = {
             type = "array",
             items = private_key_schema,
         },
+        client = {
+            type = "object",
+            properties = {
+                ca = certificate_scheme,
```
Comment (tokers) [inline apisix/schema_def.lua:648]: We may support CA cert chains in the future.

File: apisix/admin/ssl.lua [hunk 3]
```diff
@@ -70,6 +70,17 @@ local function check_conf(id, conf, need_id)
         end
     end
 
+    if conf.client then
+        if not apisix_ssl.support_client_verification() then
```
Comment (tokers) [inline apisix/admin/ssl.lua:74]: This check is difficult when we using manager-api, unless we have a mechanism to advertise this capability to it (like rolling feature sets in the server-info plugin).

File: apisix/schema_def.lua (hunk 2 above)
Comment (spacewander) [reply apisix/schema_def.lua:648]: Currently, OpenResty only supports a single CA cert chain. We can use `ca_list` or anything else if multiple chains are supported. However, using such a name now is confusing as we only support one.

File: apisix/admin/ssl.lua (hunk 3 above)
Comment (spacewander) [reply apisix/admin/ssl.lua:74]: There is an additional check in the DP side. The `conf.client` will be ignored if the DP doesn't support it. The check here is not critical as the DP may be different with the Admin API.

File: docs/en/latest/admin-api.md [hunk 4]
```diff
@@ -784,6 +784,8 @@ Return response from etcd currently.
...
 | certs       | False    | An array of certificate | when you need to configure multiple certificate for the same domain, you can pass extra https certificates (excluding the one given as cert) in this field |                                                  |
 | keys        | False    | An array of private key | https private keys. The keys should be paired with certs above                                                                                             |                                                  |
+| client.ca | False | Certificate| set the CA certificate which will use to verify client. This feature requires OpenResty 1.19+.                                                            |                                                  |
+| client.depth | False | Certificate| set the verification depth in the client certificates chain, default to 1. This feature requires OpenResty 1.19+.                                                            |                                                  |
```
Comment (Firstsawyou) [inline docs/en/latest/admin-api.md:788]: The first letter should be capitalized.  set --> Set
```
set the CA certificate which will use to verify client.
```

File: t/node/client-mtls.t [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+
+=== TEST 5: no client certificate
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin")
+            local json = require("toolkit.json")
+            local ssl_ca_cert = t.read_file("t/certs/mtls_ca.crt")
+            local ssl_cert = t.read_file("t/certs/mtls_client.crt")
+            local ssl_key = t.read_file("t/certs/mtls_client.key")
```
Comment (Firstsawyou) [inline t/node/client-mtls.t:210]: These certificates were not used in this test, I think they can be removed.

File: apisix/schema_def.lua (hunk 2 above)
Comment (tokers) [reply apisix/schema_def.lua:648]: OK, just an advice and won't block this PR.

File: apisix/admin/ssl.lua (hunk 3 above)
Comment (tokers) [reply apisix/admin/ssl.lua:74]: Got it.

File: docs/en/latest/admin-api.md (hunk 4 above)
Comment (spacewander) [reply docs/en/latest/admin-api.md:788]: None of the description in this table start with uppercase letter.

File: t/node/client-mtls.t (hunk 5 above)
Comment (spacewander) [reply t/node/client-mtls.t]: @Firstsawyou 
Updated.

File: docs/en/latest/admin-api.md (hunk 4 above)
Comment (Firstsawyou) [reply docs/en/latest/admin-api.md:788]: Got it.

**Review by tokers — APPROVED** [review]

**Pull request merged**